app/code/Magento/Inventory/Model/StockSourceLink/Command/GetAssignedSourcesForStock.php

```diff
@@ -47,17 +51,20 @@ class GetAssignedSourcesForStock implements GetAssignedSourcesForStockInterface
      * @param ResourceConnection $resourceConnection
      * @param SearchCriteriaBuilder $searchCriteriaBuilder
      * @param SourceRepositoryInterface $sourceRepository
+     * @param GetSourceLinksInterface $getSourceLinks
      * @param LoggerInterface $logger
      */
     public function __construct(
         ResourceConnection $resourceConnection,
         SearchCriteriaBuilder $searchCriteriaBuilder,
         SourceRepositoryInterface $sourceRepository,
+        GetSourceLinksInterface $getSourceLinks,
         LoggerInterface $logger
     ) {
         $this->resourceConnection = $resourceConnection;
```

Comment:
after your refactoring 
"resource connection" object not used in this class and should be eliminated.
```
        $this->resourceConnection = $resourceConnection;
```